examples/summarization/run_summarization.py

```diff
@@ -439,6 +439,16 @@ def main():
         use_auth_token=True if model_args.use_auth_token else None,
         use_cache=False if training_args.gradient_checkpointing else model_args.use_cache,
     )
+    is_bart = True if model_args.model_name_or_path in [
+            "facebook/bart-large-mnli",
+            "facebook/bart-large-cnn",
+            "facebook/bart-large",
+            "facebook/bart-base",
+            "facebook/bart-large-xsum",
+        ] else False
+    if is_bart and not (training_args.do_predict or training_args.do_eval):
+        raise ValueError("Training is not yet supported for BART. Eval or predict can be enabled")
```

Comment:
```suggestion
    is_bart = model.config.model_type == "bart"
    if is_bart and training_args.do_train:
        raise ValueError("Training is not yet supported for BART. Eval or predict can be enabled with `--do_eval` and `--do_predict`.")
```

Let's move this after model initialization at line 467 to have a more robust way of assessing whether the current model type is BART or not.